x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4

```diff
@@ -77,7 +77,7 @@ regexBooleanExpression
     ;
 
 matchBooleanExpression
-    : valueExpression MATCH queryString=string
+    : fieldExp=qualifiedName COLON queryString=primaryExpression
```

Comment:
I've simplified the grammar from the previous iteration - we allow for `qualifiedName` as field name, and `primaryExpression` instead of a `valueExpression`.

Reply:
I wonder if we can use `queryString=constant` here? I was looking for match operator tests that don't use a constant string, but I haven't found any yet. If we want to support some complicated options besides a constant, we might want to have some tests to cover them.

Reply:
I think the following could be valid query string usages:

- Function calls:

`| WHERE field: CONCAT("a query", "string)`

`| WHERE field: TO_LOWER("A QUERY STRING")`

- Casting:

`| WHERE field: ip_expression::STRING`

- Reference:

```
| EVAL query_string = ...
| WHERE field:query_string
```

I will add these to the grammar and add some tests for them, unless you think they're not necessary.

Reply:
@fang-xing-esql , I've looked into implementing the above and right now the match function does not allow for query string parameters that are not constant.

I think it's better not to address that change as part of this PR - let's work on having the equivalent syntax for the match function and the match operator, and afterwards we can work on lifting those restrictions.

Thank you for calling out the lack of tests - I now understand why those were needed in order to highlight the necessary support.

I've simplified the operator grammar to allow just for qualifiedNames as fields and constants as queries in f70f64ea110293e98a4df602ce601f86db167fb6. Let me know what you think!